### ISSUE TITLE
[react-jss] createUseStyles TypeScript typings ignores Theme generic argument for theming option

### ISSUE THREAD

**Issue by dfernandez-asapp** [cssinjs/jss#1308]
__Expected behavior:__

This code fails to compile due to type errors:
```
type MyCustomTheme = {color: string};
const ThemeContext = React.createContext<MyCustomTheme>({color: 'black'});
const theming = createTheming<MyCustomTheme>();

const useStyles = createUseStyles<MyCustomTheme>(theme => ({
   someName: {color: theme.color}
}), { 
  theming /* <-- Fails Theming<MyCustomTheme> is not assignable to Theming<object> */
});
```

__Describe the bug:__
The generic function `createUseStyles` receives a `Theme` type as a parameter: 
`createUseStyles<MyCustomTheme>`. It also receives a `theming` option to pass the implementation of `ThemeContext`. However, the `Theme` template parameter is never used in for the declaration of options.

__Codesandbox link:__ https://codesandbox.io/s/headless-night-xu428?fontsize=14&hidenavigation=1&theme=dark

__Versions (please complete the following information):__
 - jss: 10.1.1
 - TypeScript: 3.8.3
 - Browser [e.g. chrome, safari]: Chrome 
 - OS [e.g. Windows, macOS]: macOS

**Comment by kof** [comment]
Will be released in 10.1.1 or higher

**Comment by kof** [comment]
Released in v10.2.0
